frontend/lib/resolution/scope-queries.cpp

```diff
@@ -2244,6 +2262,306 @@ void resolveUsesAndImportsInScope(Context* context, const Scope* scope) {
   return;
 }
 
+// adds all elements in intersect(a,b) into the set dst
+static void doSetIntersect(const std::set<UniqueString>& a,
+                           const std::set<UniqueString>& b,
+                           std::set<UniqueString>& dst) {
+
+  std::set_intersection(a.begin(), a.end(), b.begin(), b.end(),
+                        std::inserter(dst, dst.end()),
+                        std::less<UniqueString>());
+}
+
+// adds all elements in (a-b) into the set dst
+static void doSetDifference(const std::set<UniqueString>& a,
+                            const std::set<UniqueString>& b,
+                            std::set<UniqueString>& dst) {
+
+  std::set_difference(a.begin(), a.end(), b.begin(), b.end(),
+                      std::inserter(dst, dst.end()),
+                      std::less<UniqueString>());
+}
+
+static void updateNameSets(const std::set<UniqueString>& newNames,
+                           std::set<UniqueString>& namesDefined,
+                           std::set<UniqueString>& namesDefinedMultiply) {
+  // add anything in intersect(newNames, namesDefined) to namesDefinedMultiply.
+  doSetIntersect(newNames, namesDefined, namesDefinedMultiply);
+
+  // append everything in newNames to namesDefined
+  namesDefined.insert(newNames.begin(), newNames.end());
+}
+
+static void updateNameSets2(const std::set<UniqueString>& newNames,
```

Comment:
Should be able to determine difference between this and `updateNameSets` based on name

Reply:
I changed it to just one routine (and in one case, it just passes an empty set)